proposal/color-4-new-spaces.md

```diff
@@ -0,0 +1,1645 @@
+    respectively to the results of [hue interpolation][hue-method] with
+    `channel1` as `hue1`, `channel2` as `hue2`, using the `hue-arc` method.
+
+  * Set `channel` to the result of calculating
+    `(channel1 * weight) + (channel2 * (1 - weight))`.
+
+    > Channel rounding has been removed, since it is a lossy transform.
+
+* Return `mix`.
+
+[premultiplying]: #premultiply-transparent-colors
+[un-premultiplying]: #premultiply-transparent-colors
+[color-method]: #color-interpolation-method
+[hue-method]: #hue-interpolation
+[converting]: #converting-a-color
+
+#### Premultiply Transparent Colors
+
+When the colors being interpolated are not fully opaque, they are transformed
+into premultiplied color values. This process accepts a single `color` and
+updates the channel values if necessary, returning a new color with
+premultiplied channels.
+
+* If the `color` has an `alpha` value of 1, return `color` unchanged.
+
+* Otherwise, for each `channel` in `color`:
+
+  * If either the `alpha` value, or the `channel` value is `none`, or the
+    `channel` represents a polar-angle `hue`, keep the original value of
+    `channel`.
+
+  * Otherwise, set `channel` to the result of multiplying the `channel` value
+    by the `alpha` value.
+
+* Return the resulting `color` with premultiplied channels.
+
+The same process can be run in reverse, to **un-premultiply** the channels of a
+given `color`:
+
+* If the `color` has an `alpha` value of 1, return `color` unchanged.
+
+* Otherwise, for each `channel` in `color`:
+
+  * If either the `alpha` value, or the `channel` value is 0 or `none`, or the
+    `channel` represents a polar-angle `hue`, keep the original value of
+    `channel`.
+
+  * Otherwise, set `channel` to the result of dividing the premultiplied
+    `channel` value by the `alpha` value.
+
+* Return the resulting `color` with un-premultiplied channels.
+
+#### Hue Interpolation
+
+> When interpolating between polar-angle hue channels, there are multiple
+> 'directions' the interpolation could move, following different logical rules.
+
+This process accepts two hue angles (`hue1` and `hue2`), and returns both hues
+adjusted according to the given `method`. When no hue interpolation `method` is
+specified, the default is `shorter`.
+
+The process for each [hue interpolation method][hue-interpolation] is defined
+in [CSS Color Level 4][color-4]. If the `method` is not the value
+`'specified'`, both angles need to be constrained to `[0, 360)` prior to
+interpolation.
+
+* [shorter](https://www.w3.org/TR/css-color-4/#shorter)
+
+* [longer](https://www.w3.org/TR/css-color-4/#hue-longer)
+
+* [increasing](https://www.w3.org/TR/css-color-4/#increasing)
+
+* [decreasing](https://www.w3.org/TR/css-color-4/#decreasing)
+
+* [specified](https://www.w3.org/TR/css-color-4/#hue-specified)
+
+[hue-interpolation]: https://www.w3.org/TR/css-color-4/#hue-interpolation
+
+
+## Deprecated Functions
```

Comment:
Should we also take this opportunity to deprecate the global `saturate()`, `transparentize()`, etc functions, since none of them are getting the fancy new color space logic?

Reply:
Yes. Should that also include the global versions of functions that we are updating, like `adjust-color()`?